sdk/search/azure-search-documents/src/test/java/com/azure/search/documents/TestHelpers.java

```diff
@@ -152,4 +96,77 @@ public static MatchConditions generateIfExistsAccessCondition() {
+        try {
+            assertEquals(jacksonAdapter.serialize(expected, SerializerEncoding.JSON),
+                jacksonAdapter.serialize(actual, SerializerEncoding.JSON));
+        } catch (IOException ex) {
+            fail("There is something wrong happen in serializer.");
+        }
+    }
+
+    public static void assertObjectEquals(Object expected, Object actual, boolean ignoreDefaults,
+        String ... ignoreFields) {
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode expectedNode = mapper.valueToTree(expected);
+        ObjectNode actualNode = mapper.valueToTree(actual);
+        assertOnMapIterator(expectedNode.fields(), actualNode, ignoreDefaults, ignoreFields);
+    }
+
+    private static void assertOnMapIterator(Iterator<Map.Entry<String, JsonNode>> expectedNode,
+        ObjectNode actualNode, boolean ignoreDefaults, String[] ignoreFields) {
+        while (expectedNode.hasNext()) {
+            assertTrue(actualNode.fields().hasNext());
+            Map.Entry<String, JsonNode> expectedField = expectedNode.next();
+            String fieldName = expectedField.getKey();
+            if (!doesFieldNeedCheck(fieldName, expectedField.getValue(), ignoreDefaults, ignoreFields)) {
+                continue;
+            }
+            if (expectedField.getValue().isValueNode()) {
+                assertEquals(expectedField.getValue(), actualNode.get(expectedField.getKey()));
+            } else if (expectedField.getValue().isArray()) {
+                Iterator<JsonNode> expectedArray = expectedField.getValue().elements();
+                Iterator<JsonNode> actualArray = actualNode.get(expectedField.getKey()).elements();
+                while (expectedArray.hasNext() ) {
+                    assertTrue(actualArray.hasNext());
+                    Iterator<JsonNode> expectedElements = expectedArray.next().elements();
+                    Iterator<JsonNode> actualElements = actualArray.next().elements();
+                    while (expectedElements.hasNext()) {
+                        assertTrue(actualElements.hasNext());
+                        JsonNode a = expectedElements.next();
+                        JsonNode b = actualElements.next();
+                        if (Arrays.asList(ignoreFields).contains(fieldName)) {
+                            continue;
+                        }
+                        if (!doesFieldNeedCheck(null, a, true)) {
+                            continue;
+                        }
+                        assertEquals(a.asText(), b.asText());
+                    }
+                }
+            } else {
+                assertObjectEquals(expectedField.getValue(), actualNode.get(expectedField.getKey()), ignoreDefaults, ignoreFields);
+            }
+        }
+    }
+
+    private static boolean doesFieldNeedCheck(String fieldName, JsonNode fieldValue,
+        boolean ignoreDefaults, String... ignoreFields) {
```

Comment:
If this is going to be passed in many times should we convert `ignoredFields` to a `Set` before processing?